Discovery fails with Kafka DESCRIBE_ACLS

When trying to connect to a 2.8.1 Kafka Instance I'm getting the following error:

```
com.event.discovery.agent.framework.exception.EventDiscoveryAgentException: Error getting kafka consumers
	at com.event.discovery.agent.kafkaCommon.plugin.AbstractApacheKafkaPlugin.getBrokerConfig(AbstractApacheKafkaPlugin.java:61)
	at com.event.discovery.agent.apps.eventDiscovery.EventDiscoveryApp.start(EventDiscoveryApp.java:125)
	at com.event.discovery.agent.apps.eventDiscovery.RealtimeEventDiscoveryApp.lambda$start$0(RealtimeEventDiscoveryApp.java:59)
	at com.event.discovery.agent.framework.task.RunnableJob.run(RunnableJob.java:39)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.util.concurrent.ExecutionException: org.apache.kafka.common.errors.UnsupportedVersionException: The broker does not support DESCRIBE_ACLS
	at org.apache.kafka.common.internals.KafkaFutureImpl.wrapAndThrow(KafkaFutureImpl.java:45)
	at org.apache.kafka.common.internals.KafkaFutureImpl.access$000(KafkaFutureImpl.java:32)
	at org.apache.kafka.common.internals.KafkaFutureImpl$SingleWaiter.await(KafkaFutureImpl.java:89)
	at org.apache.kafka.common.internals.KafkaFutureImpl.get(KafkaFutureImpl.java:260)
	at com.event.discovery.agent.kafkaCommon.KafkaCommon.discoveryKafkaAcls(KafkaCommon.java:397)
	at com.event.discovery.agent.kafkaCommon.KafkaCommon.buildKafkaDiscoveryData(KafkaCommon.java:202)
	at com.event.discovery.agent.kafkaCommon.plugin.AbstractApacheKafkaPlugin.getBrokerConfig(AbstractApacheKafkaPlugin.java:58)
	... 6 common frames omitted
Caused by: org.apache.kafka.common.errors.UnsupportedVersionException: The broker does not support DESCRIBE_ACLS

Exception in thread "EDA-1" com.event.discovery.agent.framework.exception.EventDiscoveryAgentException: Error getting kafka consumers
	at com.event.discovery.agent.framework.task.RunnableJob.handleException(RunnableJob.java:53)
	at com.event.discovery.agent.framework.task.RunnableJob.run(RunnableJob.java:41)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834
```

--- a/service/libraries/kafkaCommon/src/main/java/com/event/discovery/agent/kafkaCommon/KafkaCommon.java
+++ b/service/libraries/kafkaCommon/src/main/java/com/event/discovery/agent/kafkaCommon/KafkaCommon.java
@@ -41,6 +41,7 @@ import org.apache.kafka.common.acl.AclBindingFilter;
 import org.apache.kafka.common.config.ConfigResource;
 import org.apache.kafka.common.errors.ClusterAuthorizationException;
 import org.apache.kafka.common.errors.SecurityDisabledException;
+import org.apache.kafka.common.errors.UnsupportedVersionException;
 import org.springframework.beans.factory.ObjectProvider;
 import org.springframework.beans.factory.config.BeanDefinition;
 import org.springframework.context.annotation.Scope;
@@ -400,6 +401,8 @@ public class KafkaCommon {
                 log.info("No Authorizer is configured on the broker");
             } else if (e.getCause() instanceof ClusterAuthorizationException) {
                 log.info("User '" + brokerAuthentication.getAdminUsername() + "' is not authorized to discover ACLs");
+            } else if (e.getCause() instanceof UnsupportedVersionException) {
+                log.info("Ignoring ACLs." + e.getMessage());
             } else {
                 throw e;
             }

--- a/service/services/eventDiscoveryService/install/offline/awsmsk/build.sh
+++ b/service/services/eventDiscoveryService/install/offline/awsmsk/build.sh
@@ -1,3 +1,3 @@
-cp ../../../target/event-discovery-agent-1.0-SNAPSHOT.jar .
-docker build --build-arg JAR_FILE=event-discovery-agent-1.0-SNAPSHOT.jar --tag event-discovery-agent:1.0 .
-rm ./event-discovery-agent-1.0-SNAPSHOT.jar
+cp ../../../target/event-discovery-agent-app-1.0-SNAPSHOT.jar .
+docker build --build-arg JAR_FILE=event-discovery-agent-app-1.0-SNAPSHOT.jar --build-arg AGENT_VERSION=1.0.0 --tag async-api-tool:1.0 .
+rm ./event-discovery-agent-app-1.0-SNAPSHOT.jar

--- a/service/services/eventDiscoveryService/src/test/java/com/event/discovery/agent/integrationTests/offline/kafkaCommon/kafka/KafkaEventDiscoveryTestsIT.java
+++ b/service/services/eventDiscoveryService/src/test/java/com/event/discovery/agent/integrationTests/offline/kafkaCommon/kafka/KafkaEventDiscoveryTestsIT.java
@@ -22,6 +22,7 @@ import org.apache.kafka.common.KafkaFuture;
 import org.apache.kafka.common.acl.AclBinding;
 import org.apache.kafka.common.errors.ClusterAuthorizationException;
 import org.apache.kafka.common.errors.SecurityDisabledException;
+import org.apache.kafka.common.errors.UnsupportedVersionException;
 import org.apache.kafka.common.internals.KafkaFutureImpl;
 import org.hamcrest.CoreMatchers;
 import org.junit.Test;
@@ -369,6 +370,27 @@ public class KafkaEventDiscoveryTestsIT extends KafkaCommonTestsIT {
 
         final JobDTO status = getDiscoveryOperationStatus(jobId);
         assertThat(status.getStatus(), is(JobStatus.ERROR));
+    }
+
+    @Test
+    public void testAclUnsupportedVersionHandledGracefully() throws Exception {
+        KafkaFuture<Collection<AclBinding>> future = new KafkaFutureImpl<>() {
+            @Override
+            public Collection<AclBinding> get() throws InterruptedException, ExecutionException {
+                throw new ExecutionException(new UnsupportedVersionException("The broker does not support DESCRIBE_ACLS"));
+            }
+        };
+
+        kafkaAdminClientMocks.configureDescribeAclResult(future);
+
+        DiscoveryOperationResponseDTO response = createEventListenerOperation(
+                kafkaConfigRequests.createDefaultKafkaEventDiscoveryRequest());
+
+        final String jobId = response.getJobId();
+        assertNotNull(jobId);
+
+        final JobDTO status = getDiscoveryOperationStatus(jobId);
+        assertThat(status.getStatus(), is(JobStatus.COMPLETED));
     }
 
     private void setSSLExpectedClientProperties(Map<String, Object> adminClientProps) {

